K8SPXC-489 Operator file bundle for GCP (#52)
percona/percona-helm-charts: charts/gcp-marketplace/pxc-operator/README.md

Create a new cluster from the command line:

```shell
export CLUSTER=pxc-cluster
export ZONE=us-west1-a

gcloud container clusters create "$CLUSTER" --zone "$ZONE"
```

Configure `kubectl` to connect to the new cluster.

```shell
gcloud container clusters get-credentials "$CLUSTER" --zone "$ZONE"
```

#### Clone this repo

Clone this repo and the associated tools repo:

```shell
git clone --recursive https://github.com/percona/percona-helm-charts
```

#### Install the Application resource definition

An Application resource is a collection of individual Kubernetes components,
such as Services, Deployments, and so on, that you can manage as a group.

To set up your cluster to understand Application resources, run the following command:

```shell
kubectl apply -f "https://raw.githubusercontent.com/GoogleCloudPlatform/marketplace-k8s-app-tools/master/crd/app-crd.yaml"
```

You need to run this command once.

The Application resource is defined by the
[Kubernetes SIG-apps](https://github.com/kubernetes/community/tree/master/sig-apps) community. The source code can be found on
[github.com/kubernetes-sigs/application](https://github.com/kubernetes-sigs/application).

> ### Install Percona Kubernetes Operator for Percona Server for MongoDB

Reviewer: ```suggestion
### Install Percona Kubernetes Operator for Percona XtraDB Cluster
```

Author: done

Navigate to the `pxc-operator` directory:

```shell
cd percona-helm-charts/charts/gcp-marketplace/pxc-operator
```

#### Configure the environment variables

Choose an instance name and
[namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)
for the app. In most cases, you can use the `default` namespace.

```shell
export APP_INSTANCE_NAME=pxc-operator
export NAMESPACE=default
```

Set up the image tag:

It is advised to use stable image reference which you can find on

> [Marketplace Container Registry](https://marketplace.gcr.io/google/psmdb-operator).

Author: done

```

To view the app, open the URL in your browser.

### Access Percona XtraDB Cluster within the network

You can connect to Percona XtraDB Cluster without exposing it to public access, using the
`mysql` command line interface. We recommend making all connection via the corresponding network service.

The user credentials are configured on application start up by the operator. More information about user management
could be found [here](https://www.percona.com/doc/kubernetes-operator-for-pxc/users.html)

#### Connect to Percona XtraDB Cluster using a client Pod

Run percona-client and connect its console output to your terminal (running it may require some time to deploy the correspondent Pod):

```shell
kubectl run -i --rm --tty percona-client --image=percona:8.0 --restart=Never \
            --env MYSQL_PASSWORD=$(kubectl get secret/${APP_INSTANCE_NAME}-secrets -o jsonpath='{.data.root}' | base64 -d) \
            --env NAMESPACE=${NAMESPACE} \
            --env NAME=${APP_INSTANCE_NAME} \
            -- bash -il
#wait for shell and than run
mysql -h ${NAME}-haproxy -uroot -p${MYSQL_PASSWORD}
```

# Scaling

## Scaling the cluster up or down

> By default, Percona Server for MongoDB application is deployed using 3 replicas.

Reviewer: ```suggestion
By default, Percona XtraDB Cluster application is deployed using 3 replicas.
```

Author: done

To change the number of replicas, use the following command:

```
kubectl patch psmdb "${APP_INSTANCE_NAME}" \
  --namespace "${NAMESPACE}" \
  --type=json \

>   -p '[{"op":"replace","path":"/spec/replsets/0/size","value":'${REPLICAS}'}]'

Reviewer: This is for PSMDB again.

Author: done

```

Where `REPLICAS` is the number of replicas you want.

# Backup and Restore

> The detailed set of steps for backup and restores you can find here [Backups and Restores documentation](https://www.percona.com/doc/kubernetes-operator-for-psmongodb/backups.html).

Reviewer: ```suggestion
The detailed set of steps for backup and restores you can find here [Backups and Restores documentation](https://www.percona.com/doc/kubernetes-operator-for-pxc/backups.html).
```

Author: done

# Uninstall the Application

## Using the Google Cloud Platform Console

1. In the GCP Console, open [Kubernetes Applications](https://console.cloud.google.com/kubernetes/application).

> 1. From the list of applications, click **Percona Server MongoDB Operator**.

Reviewer: Not sure how it would look for PXC, but needs to be changed.

Author: done

1. On the Application Details page, click **Delete**.

## Using the command line

### Prepare the environment

Set your installation name and Kubernetes namespace:

```shell

> export APP_INSTANCE_NAME=psmdb-operator

Author: done
